client-course-schedulizer/src/utilities/hooks/useLocal.ts

```diff
@@ -7,7 +7,12 @@
  */
 export const loadLocal = <T>(key: string) => {
   const item: string | null = localStorage.getItem(key);
-  return item ? (JSON.parse(item) as T) : undefined;
+  const thing = item ? JSON.parse(item) : undefined;
+  // Clear fileUrl
+  if (thing !== undefined) {
+    thing.fileUrl = "";
+  }
+  return thing ? (thing as T) : undefined;
```

Comment:
This is good, except that `loadLocal` is a generic function that can't guarantee T will have the property `fileUrl`. This error can be seen by typing `thing` as `T | undefined` --> `const thing: T | undefined = item ? JSON.parse(item) : undefined;`. `loadLocal` is good for loading values from localStorage but assigning values to the object from storage is best done in another function. For example, create a function similar to the code below, which would be used in `appInterfaces.ts` in place of `loadLocal("appState")` 


```tsx
const loadInitalState = (): AppState | undefined => {
  const savedState = loadLocal<AppState>("appState");
  if (savedState !== undefined) {
    savedState.fileUrl = "";
  }
  return savedState;
}
```

Reply:
Thanks! 6d902b1